### ISSUE TITLE
Support `respect-type-ignore-comments` in file overrides

### ISSUE THREAD

**Issue by strokirk** [astral-sh/ty#2530]
Hello!

https://github.com/astral-sh/ty/issues/1422 and https://github.com/astral-sh/ruff/pull/22137, which added `respect-type-ignore-comments` as a global configuration option, have been very useful for me to avoid warnings in ty where it doesn't agree with mypy. I'd love if that could be applied on a file-by-file basis using overrides. 

A common example is Pydantic's `@computed_field` decorator, which requires `# type: ignore[prop-decorator]` for mypy but doesn't need any ignore comment for ty.

Since a lot of those models are usually defined in a single file, or file matching a common pattern, an override would be super useful.

## Example

```python
from pydantic import BaseModel, computed_field

class MyModel(BaseModel):
    @computed_field  # type: ignore[prop-decorator]
    @property
    def my_prop(self) -> dict:
        ...
```

Thanks for a great tool!

**Comment by MichaReiser** [comment]
Thanks for opening this issue. This shouldn't be too hard to add. We only waited with adding support for it because we first wanted to hear from users what use cases they have for it. Now we know! Thank you
